### ISSUE TITLE
Scraping randomly stops after few hours

### ISSUE THREAD

**Issue by spdesmith** [isaackogan/TikTokLive#137]
**Describe the bug**
If I remove ethernet it does stop for a bit and then reconnects which is very useful so I don't think this is a disconnect issue, but after a while it just stops, and I'm not sure why. Any idea how we could fix it?

**Expected behavior**
I need it to just run for long hours and stop when I close it

**Desktop (please complete the following information):**
- chrome
- windows 11

**To Reproduce**

``` import socket
import asyncio
from TikTokLive import TikTokLiveClient
from TikTokLive.types.events import ConnectEvent, CommentEvent, GiftEvent, LikeEvent, JoinEvent, FollowEvent, ShareEvent

# Create a UDP socket
sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)

# Specify the IP address and port number to send the UDP packets to
server_address = ('localhost', 12345)  # replace with your IP and port

class Tiktok:

    # Initialize the Tiktok class with a user ID
    def __init__(self, user):
        self.user = user  # Store the user ID
        # Create a new event loop specific for this user
        self.loop = asyncio.new_event_loop()
        # Set a custom exception handler for the loop (to handle any errors during async operations)
        self.loop.set_exception_handler(self.exception_handler)
        # Initialize the TikTok live client with the given user ID, the created event loop, and enable detailed gifts
        self.client = TikTokLiveClient(unique_id=user, loop=self.loop, enable_detailed_gifts=True)

    async def connect(self):
        while True:
            if self.client.connected:
                #print('connected:', self.user)
                await asyncio.sleep(5)
            if self.client.connecting:
                print('connecting:', self.user)
                await asyncio.sleep(5)
            elif not self.client.connected and not self.client.connecting:
                print('not connect: ', self.user)
                try:
                    await self.client.start()
                except LiveNotFound:
                    self.client.stop()
                    await asyncio.sleep(240)

    def exception_handler(self, loop, context):
        print("Exception occurred:", context)

tiktok_client = Tiktok("@kai_cenat")

@tiktok_client.client.on("connect")
async def on_connect(_: ConnectEvent):
    print("Connected to room ID", tiktok_client.client.room_id)

@tiktok_client.client.on("comment")
async def on_comment(event: CommentEvent):
    comment_message = f"@{event.user.unique_id}|commented|{event.comment}\n".encode("utf-8")
    sock.sendto(comment_message, server_address)
    print(comment_message)

async def start(loop):
    asyncio.run_coroutine_threadsafe(tiktok_client.connect(), loop)
    while True:
        await asyncio.sleep(10)

if __name__ == '__main__':
    loop = asyncio.new_event_loop()
    loop.run_until_complete(start(loop))```

**Comment by Xr-TenderLIfe** [comment]
The same problem,After connecting for a period of time, a fully open link without warning will not throw an exception and the state will not change.

**Comment by Xr-TenderLIfe** [comment]
After connecting for a period of time, a fully open link without warning will not throw an exception and the state will not change.

**Comment by atyrode** [comment]
I wonder if this is related to this issue I had last year and reported along with someone else on the discord:
https://discord.com/channels/977648006063091742/978700446279565322/1087090180835508264

Below this message you'll find a discussion that highlights the same issue, after some digging back then I seemed to have pinpointed the issue, but sadly the fix doesn't work now anymore, so I'm going through the debugging of this issue again.

If it is the same, then it relates to webcast error handling/raising, I'm lead to believe.

**Comment by isaackogan** [comment]
> I wonder if this is related to this issue I had earlier this year and reported along with someone else on the discord: https://discord.com/channels/977648006063091742/978700446279565322/1087090180835508264
> 
> Below this message you'll find a discussion that highlights what I think might be the same issue, after some digging back then I seemed to have pinpointed the issue, but sadly the fix doesn't work now anymore, so I'm going through the debugging of this issue again.
> 
> If it is the same, then it relates to webcast error handling/raising, I'm lead to believe. I think my crash is not relating to this library, but isn't properly raised due to some async behavior of the library itself, resulting in my crash never being properly handled on my side.

I'm not sure async behaviour of the library causes the error to not be raised...what is this conjecture based off of? I believe TikTok just stops sending events. This sentiment seems to be echoed by the other libraries.

**Comment by isaackogan** [comment]
@zerodystrash suggests this might be due to an ack not being received/accepted by TikTok. Will be looking into this further in the next version.

Please see https://github.com/isaackogan/TikTokLive/issues/157 for information.

**Comment by valmir123** [comment]
Hi @isaackogan 

I'm also getting the same issue where scraping suddenly stops without giving any error.

Any chance you've got some insights on what might be causing this?

Aside from this issue, the library has been functioning flawlessly for me :)

Thank you for the effort!

Kind regards
Valmir

**Comment by isaackogan** [comment]
It's something I've seen and am fixing

**Comment by isaackogan** [comment]
Fixed in v6.0.0!

**Comment by valmir123** [comment]
> Fixed in v6.0.0!

Thank you.

What was the issue exactly and how did you fix it if I may ask?

**Comment by isaackogan** [comment]
We were not sending the correct payload to TikTok to acknowledge requests, which means after a while TikTok would assume we were a broken client and stop sending events.
